File: src/main/java/club/xiaojiawei/entity/area/GraveyardArea.java
package club.xiaojiawei.entity.area;

import club.xiaojiawei.entity.Card;

/**
 * @author 肖嘉威
 * @date 2022/11/27 15:02
 */
public class GraveyardArea extends Area {

    public GraveyardArea() {
        super(Integer.MAX_VALUE);
    }

    @Override
    public void putZeroAreaCard(Card card) {
        add(card);
    }

}

File: hs-script-card-sdk/src/main/kotlin/club/xiaojiawei/bean/area/GraveyardArea.kt
package club.xiaojiawei.bean.area

import club.xiaojiawei.bean.Card
import club.xiaojiawei.bean.Player

/**
 * 墓地
 * @author 肖嘉威
 * @date 2022/11/27 15:02
 */
class GraveyardArea(allowLog: Boolean = false, player: Player) :
    Area(allowLog = allowLog, maxSize = Int.MAX_VALUE, player = player) {

    override fun addZeroCard(card: Card?) {
        add(card)
    }

    override fun cardSize(): Int {
        return zeroCardSize()
    }
}
